project-management-openproject/project-management-openproject-api/src/main/java/com/xwiki/projectmanagement/openproject/internal/processing/OpenProjectFilterHandler.java

```diff
@@ -50,89 +52,195 @@ private OpenProjectFilterHandler()
+        for (LiveDataQuery.Constraint constraint : filterConstraints) {
+            String operatorValue = OpenProjectMapper.mapLivedataOperator(constraint.getOperator());
+
+            if (result.containsKey(operatorValue)) {
+                List<String> currentValues = (List<String>) result.get(operatorValue);
+                currentValues.add((String) constraint.getValue());
+                result.put(operatorValue, currentValues);
+            } else {
+                List<String> values = new ArrayList<String>(
+                    Collections.singletonList((String) constraint.getValue()));
+                result.put(operatorValue, values);
+            }
+        }
+        return result;
+    }
+
+    private static String convertMapOfFiltersToString(Map<String, Object> filters) throws ProjectManagementException
+    {
+        List<Map<String, Object>> convertedFilters = new ArrayList<>();
+
+        for (Map.Entry<String, Object> entry : filters.entrySet()) {
+            Map<String, Object> constraints = (Map<String, Object>) entry.getValue();
+            constraints.forEach(
+                (key, value) -> {
+                    convertedFilters.add(
+                        Map.of(
+                            entry.getKey(),
+                            Map.of(
+                                OPERATOR, key,
+                                VALUES, value
+                            )
+                        )
+                    );
+                }
+            );
+        }
+        return convertListUsingMapper(convertedFilters);
+    }
+
+    private static void mergeFiltersMaps(Map<String, Object> firstMap, Map<String, Object> secondMap)
+    {
+        for (Map.Entry<String, Object> entry : secondMap.entrySet()) {
+            if (firstMap.containsKey(entry.getKey())) {
+                Map<String, Object> firstFilter = (Map<String, Object>) firstMap.get(entry.getKey());
+                Map<String, Object> secondFilter = (Map<String, Object>) entry.getValue();
+                secondFilter.forEach((key, value) -> {
+                    if (firstFilter.containsKey(key)) {
+                        List<String> firstFilterValues = (List<String>) firstFilter.get(key);
+                        List<String> secondFilterValues = (List<String>) value;
+                        for (String filterValue : secondFilterValues) {
+                            if (!firstFilterValues.contains(filterValue)) {
+                                firstFilterValues.add(filterValue);
+                            }
+                        }
+                    } else {
+                        firstFilter.put(key, value);
+                    }
+                });
+            } else {
+                firstMap.put(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    private static String convertListUsingMapper(List<Map<String, Object>> filters) throws ProjectManagementException
```

Comment:
Duplicate method. This does the same thing as `#convertListOfSortEntriesToString`